xscen/regridding.py

```diff
@@ -52,105 +71,131 @@ def regrid(
+        # if this is not the first iteration (out != None),
+        # get result from last iteration (out) as input
+        ds = out or ds
+
+        # Whether or not regridding is required
+        if ds["lon"].equals(ds_grid["lon"]) & ds["lat"].equals(ds_grid["lat"]):
+            out = ds
+            if "mask" in out:
+                out = out.where(out.mask == 1)
+                out = out.drop_vars(["mask"])
 
-    else:
-        kwargs = deepcopy(regridder_kwargs)
-        # if weights_location does no exist, create it
-        if not os.path.exists(weights_location):
-            os.makedirs(weights_location)
-        id = ds.attrs["cat/id"] if "cat/id" in ds.attrs else "weights"
-        # give unique name to weights file
-        weights_filename = os.path.join(
-            weights_location,
-            f"{id}_"
-            f"{'_'.join(kwargs[k] for k in kwargs if isinstance(kwargs[k], str))}.nc",
-        )
+        else:
+            kwargs = deepcopy(regridder_kwargs)
+            # if weights_location does no exist, create it
+            if not os.path.exists(weights_location):
+                os.makedirs(weights_location)
+            id = ds.attrs["cat/id"] if "cat/id" in ds.attrs else "weights"
+            # give unique name to weights file
+            weights_filename = os.path.join(
+                weights_location,
+                f"{id}_regrid{i}"
+                f"{'_'.join(kwargs[k] for k in kwargs if isinstance(kwargs[k], str))}.nc",
+            )
 
-        # TODO: Support for conservative regridding (use xESMF to add corner information), Locstreams, etc.
+            # TODO: Support for conservative regridding (use xESMF to add corner information), Locstreams, etc.
 
-        # Re-use existing weight file if possible
-        if os.path.isfile(weights_filename) and not (
-            ("reuse_weights" in kwargs) and (kwargs["reuse_weights"] is False)
-        ):
-            kwargs["weights"] = weights_filename
-            kwargs["reuse_weights"] = True
-        regridder = _regridder(
-            ds_in=ds, ds_grid=ds_grid, filename=weights_filename, **regridder_kwargs
-        )
+            # Re-use existing weight file if possible
+            if os.path.isfile(weights_filename) and not (
+                ("reuse_weights" in kwargs) and (kwargs["reuse_weights"] is False)
+            ):
+                kwargs["weights"] = weights_filename
+                kwargs["reuse_weights"] = True
+            regridder = _regridder(
+                ds_in=ds, ds_grid=ds_grid, filename=weights_filename, **regridder_kwargs
+            )
 
-        # The regridder (when fed Datasets) doesn't like if 'mask' is present.
-        if "mask" in ds:
-            ds = ds.drop_vars(["mask"])
-        out = regridder(ds, keep_attrs=True)
+            # The regridder (when fed Datasets) doesn't like if 'mask' is present.
+            if "mask" in ds:
+                ds = ds.drop_vars(["mask"])
```

Comment:
I _think_ this is fine, but is there a problem with dropping `mask` in the first loop?

Reply:
as discussed, I think it is okay to not have masks for the intermediate grids.